File: tiq-e/array/2/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit: i32 = 0;

        for i in 1..prices.len() {
            if prices[i] > prices[i - 1] {
                profit += prices[i] - prices[i - 1]
            }
        }

        return profit;
    }
}

fn main() {
    let prices = Vec::from([7, 1, 5, 3, 6, 4]);

    let start = Instant::now();
    let result = Solution::max_profit(prices);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}

File: tiq-e/array/2/main.go
package main

import (
	"fmt"
	"time"
)

func maxProfit(prices []int) int {
	profit := 0

	for i := 1; i < len(prices); i++ {
		if prices[i] > prices[i-1] {
			profit += prices[i] - prices[i-1]
		}
	}

	return profit
}

func main() {
	var prices = []int{7, 1, 5, 3, 6, 4}

	var start = time.Now()
	var result = maxProfit(prices)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
